### PR TITLE
Isssue 79/disable api coverage in prod

### LINKED ISSUE #79: Disable /api/coverage in production
When `NODE_ENV` is `'production'`, i.e. the site is running live in production, the `/api/coverage` route should return 404.

### PR DESCRIPTION
**Pull request by kjersti**
If I understood the issue correctly, this should close #79

### CODE REVIEW

**Review by olleolleolle — APPROVED** [review]
👍 I am guessing that NODE_ENV production (aka "deployed somewhere") is the right thing here, since code coverage is only interesting in dev/testing environments.

Kudos!

File: .env.test [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # Client settings
 NEXT_PUBLIC_APP_HOST=localhost:3000
 NEXT_PUBLIC_APP_USE_TLS=0
+NEXT_PUBLIC_CYPRESS=true
```
Comment (richardolsson) [inline .env.test:4]: I would rename this to `CYPRESS` and set it's value to `1` for consistency (compare with `NEXT_PUBLIC_APP_USE_TLS`.

The `NEXT_PUBLIC_` prefix means that the variable will be compiled into the front-end bundle, i.e. be exposed to the browser. Although there is no security issue in exposing this particular variable to the browser, I think it only needs to be available to server-side code and we should avoid creating a pattern of exposing more than necessary.

File: src/pages/api/coverage.ts [hunk 2]
```diff
@@ -4,12 +4,12 @@ import { NextApiRequest, NextApiResponse } from 'next';
...
         res.status(404);
     }
     else {
         res.status(200).json({
             coverage: globalAny.__coverage__ || null,
         });
     }
-};
\ No newline at end of file
+};
```
Comment (richardolsson) [inline src/pages/api/coverage.ts:15]: It looks like your editor has added a trailing newline in the file. This is not a big problem in itself, but it tells me that maybe your editor has not picked up the settings from the `.editorconfig` file. See the README for instructions/links on how to set that up. 😊

**Review by richardolsson — CHANGES_REQUESTED** [review]
Great work! 💎 

I'm really impressed with how quickly you familiarized yourself with the environment and was able to sort this out.

I have a few comments below, and hope you can take a look at them before we merge?

**Comment by kjersti** [comment]
Let me know if there are any other issues! :)

File: src/pages/api/coverage.ts [hunk 3]
```diff
@@ -1,11 +1,15 @@
...
 
-export default (req : NextApiRequest, res : NextApiResponse) : void => {
-    // TODO: Disable this in production
-    res.status(200).json({
-        coverage: globalAny.__coverage__ || null,
-    });
-};
\ No newline at end of file
+export default (req: NextApiRequest, res: NextApiResponse): void => {
+    if (process.env.NODE_ENV === 'production' && !process.env.NEXT_PUBLIC_CYPRESS) {
```
Comment (richardolsson) [inline src/pages/api/coverage.ts:7]: The name of the variable needs to be changed here as well as a consequence of the change in 7fba7130dc4a80533dc38959d7960f6bd9d8bba3.

**Review by richardolsson — CHANGES_REQUESTED** [review]
Thanks for changing this @kjersti! The environment variable is still referenced by it's old name in the source code so it needs to be changed there as well. See comment below. 👇

**Comment by richardolsson** [comment]
Oh it looks like we found the same flaw. 😉 

I'm rerunning the CI tests and if all is well, will merge! Thank you @kjersti

**Review by richardolsson — APPROVED** [review]
Everything is looking great now! 💎 

Congratulations (and thank you) on your first merged pull request, @kjersti. 🙌

**Pull request merged**